packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/data_type/mod.rs

```diff
@@ -1,15 +1,44 @@
 pub mod resolve;
 
+use std::collections::{hash_map::Entry, HashMap};
+
 use async_trait::async_trait;
 use error_stack::{IntoReport, Result, ResultExt};
+use futures::{stream, StreamExt, TryStreamExt};
 use tokio_postgres::GenericClient;
-use type_system::DataType;
+use type_system::{uri::VersionedUri, DataType};
 
 use crate::{
-    ontology::{AccountId, PersistedOntologyIdentifier},
-    store::{AsClient, DataTypeStore, InsertionError, PostgresStore, UpdateError},
+    ontology::{AccountId, DataTypeTree, PersistedDataType, PersistedOntologyIdentifier},
+    store::{
+        crud::Read,
+        postgres::{context::PostgresContext, PersistedOntologyType},
+        query::Expression,
+        AsClient, DataTypeStore, InsertionError, PostgresStore, QueryError, UpdateError,
+    },
 };
 
+impl<C: AsClient> PostgresStore<C> {
+    /// Internal method to read a [`PersistedDataType`] into a [`HashMap`].
+    ///
+    /// This is used to recursively resolve a type, so the result can be reused.
+    pub(crate) async fn get_data_type_as_dependency(
+        &self,
+        data_type_uri: VersionedUri,
+        data_types: &mut HashMap<VersionedUri, PersistedDataType>,
+        _data_type_resolve_depth: u8,
```

Comment:
Should we add a small comment (on both methods) saying this is unused until data types can reference other data types?

Relevant asana I believe would be this https://app.asana.com/0/1200211978612931/1202464168422955/f

Reply:
Added a comment: 89527b4e786de847840403b1971355b563514f51 (due to the above changes, only one comment was enough)
I added a similar comment (without TODO) on link types in 9a99c6a5b2232b4fe233eb692f98253b9036997c